### ISSUE TITLE
The callback of onClose can be executed when you click any way

### ISSUE THREAD

**Issue by dgczhh** [iamhosseindhv/notistack#220]
I have two problems on notistack.
First, I set up the callback of onClose on a snackbar and show it, the callback will be executed when I click any way.
Second, when I add a close button on snackbar to close it manually,the callback of onClose will not be executed.
I fixed the two problem temporarily by adding some code to the source code of SnackbarItem.js and SnackbarProvider.js.

## Expected Behavior

When I click any way ,the callback of onClose on a snackbar can't be executed.
When I close a snackbar manually, the callback of onClose on a snackbar can be executed.

## Current Behavior

Contrary to the Expected Behavior.
## Steps to Reproduce

Link: https://github.com/dgczhh/notistackdemo
1.npm install
2.npm start 
3.open the Chrome DevTools
4.click the button in web and check the output

## Context

I fixed the first problem temporarily by adding some code to `node_modules\notistack\build\SnackbarItem\SnackbarItem.js` after line 78:
```
if(reason === _constants.REASONS.CLICKAWAY) {
     return;
}
```
I fixed the second problem temporarily by adding some code to `node_modules\notistack\build\SnackbarProvider.js` after line 217:
```
  if(item.key === key) {
         if(reason === null) {
                 if(item.onClose) {
                          item.onClose()
                  }
          }
     }
```

## Your Environment

| Tech         | Version |
|--------------|---------|
| Notistack | ^0.9.7  |
| React        |   ^16.12.0      |
| Browser      |     Chrome 79.03945.130    |
| etc.         |         |

**Comment by iamhosseindhv** [comment]
Regarding your first problem, it's not really a problem, but expected behaviour. Notistack also does nothing on `clickaway` event:
https://github.com/iamhosseindhv/notistack/blob/38f598c2f9b0be39a3cb9168ae85b0edb1eeb0b2/src/SnackbarProvider.js#L190
You can add this to your `onClose` callbacks too.

Second problem, you're right. when a snackbar is closed using `closeSnackbar` method, `onClose` callback passed in options parameter of `enqueueSnackbar` does not get called.

**Comment by iamhosseindhv** [comment]
Fix will be published in `v0.9.8`

**Comment by dgczhh** [comment]
Thanks a lot!
